### PR TITLE
Add ability to specify Testcafe Configuration Path

### PR DESCRIPTION
**Pull request by Chris-Greaves**
## Purpose
Add the ability to specify the Testcafe config to use, instead of the default `.testcaferc.json` file. This is very useful for Monorepos, Pipelines and QoL for developers.

## Approach
I added this functionality by extending the object passed into the `createTestCafe` method and adding checks to ensure the addition doesn't cause a breaking change.

## References
This is a solution to issue #3428 

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

**Review by miherlosev — CHANGES_REQUESTED** [review]
Thank you for contributing to the TestCafe. In general, your PR is great. I left some comments related to code style and formatting.

**Comment by Chris-Greaves** [comment]
@miherlosev Will look at making these changes this evening ready for tomorrow. Also need to update the docs so will do that.

File: test/server/configuration-test.js [hunk 1]
```diff
@@ -640,4 +641,60 @@ describe('TypeScriptConfiguration', function () {
...
+
+    describe('Custom Testcafe Config Path', () => {
+        let configuration;
+
+        afterEach(async () => {
+            await del([configuration.filePath]);
+        });
+
+        it('Custom config path is used', () => {
+            const customConfigPath = 'custom11.testcaferc.json';
```
[the remark this answers is not in the document]
Comment (Chris-Greaves) [reply test/server/configuration-test.js:653]: Did you want the new line between 652 and 653, or between 653 and 654?

[the remark this answers is not in the document]
Comment (miherlosev) [reply test/server/configuration-test.js:653]: Between 653 and 654

File: src/cli/argument-parser.ts [hunk 2]
```diff
@@ -71,6 +71,7 @@ interface CommandLineOptions {
     videoOptions?: string | Dictionary<number | string | boolean>;
     videoEncodingOptions?: string | Dictionary<number | string | boolean>;
     compilerOptions?: string | Dictionary<number | string | boolean>;
+    configPath?: string;
```
[the remark this answers is not in the document]
Comment (Chris-Greaves) [reply src/cli/argument-parser.ts:74]: Do you want me to change the cli flag to `--config-file` as well?

**Comment by Chris-Greaves** [comment]
@miherlosev Your comments should be resolved now, and the docs have been updated to include the new functionality. Any problems let me know.

**Comment by Chris-Greaves** [comment]
Looks like the "continuous-integration-azure-mobile" check failed due to it taking too long, I'm guessing I can leave that to you guys to re-run or whatever?

**Review by miherlosev — APPROVED** [review]

**Comment by miherlosev** [comment]
https://github.com/DevExpress/testcafe/pull/6035

> Looks like the "continuous-integration-azure-mobile" check failed due to it taking too long, I'm guessing I can leave that to you to re-run or whatever?

Yes, we will rerun this task.

@DIRECTcut check please documentation and option CLI description

**Comment by DIRECTcut** [comment]
@Chris-Greaves 

Hello and thank you for this awesome PR!

As our [contribution guidelines](https://github.com/DevExpress/testcafe/blob/master/CONTRIBUTING.md#documentation-for-new-features) suggest, documentation for new features should be committed to the `new-docs` branch. This is necessary to keep our website ready to publish at all times.

Please transfer all documentation changes to a new branch and open a pull request from there.
When the new PR is ready, feel free to request a review from me. Please open the branch for commits (I may need to commit some edits).

Thanks.

**Comment by Chris-Greaves** [comment]
@DIRECTcut I have created a new PR (#6053) for the docs and have referenced this outstanding PR. It should be open for you to make changes as well if needed. I have also reverted the changes to the docs from this PR so it isn't duplicated.

Just let me know if there is anything else you need me to do 😄

**Review by miherlosev — APPROVED** [review]

File: src/cli/argument-parser.ts [hunk 3]
```diff
@@ -149,6 +150,7 @@ export default class CLIArgumentParser {
             .option('--dev', 'enables mechanisms to log and diagnose errors')
             .option('--qr-code', 'outputs QR-code that repeats URLs used to connect the remote browsers')
             .option('--sf, --stop-on-first-fail', 'stop an entire test run if any test fails')
+            .option('--config-file <path>', 'use a different testcafe config to the default and specify its location')
```
Comment (DIRECTcut) [inline src/cli/argument-parser.ts:153]:
```suggestion
            .option('--config-file <path>', 'specify a custom path to the testcafe configuration file')
```

**Review by DIRECTcut — APPROVED** [review]

**Comment by Chris-Greaves** [comment]
@miherlosev I noticed that my branch is something like 13 commits behind master, did you need me to rebase the changes or anything?

**Comment by miherlosev** [comment]
All tests are green
* all tasks are green (except few `Travis CI` tasks) https://github.com/DevExpress/testcafe/pull/6035#issuecomment-801017080
* all Travis CI tasks are green https://travis-ci.com/github/testcafe-build-bot/testcafe/builds/220355952

**Comment by miherlosev** [comment]
The feature will be shipped with the next TestCafe version. We are going to release it in a few days.

**Pull request merged**